common/djangoapps/third_party_auth/management/commands/saml.py

```diff
@@ -76,45 +72,146 @@ def _handle_pull_metadata(self):
         provider_configs = SAMLProviderConfig.objects.current_set()
         if site_id:
             provider_configs = provider_configs.filter(site_id=site_id)
 
+        self.stdout.write(self.style.SUCCESS("SAML Configuration Check Report"))
+        self.stdout.write("=" * 50)
+
         for provider_config in provider_configs:
-            if provider_config.saml_configuration:
-                try:
-                    current_config = SAMLConfiguration.current(
-                        provider_config.site_id,
-                        provider_config.saml_configuration.slug
-                    )
+            total_providers += 1
+            provider_info = (
+                f"Provider '{provider_config.slug}' "
+                f"(ID: {provider_config.id}, site {provider_config.site_id})"
+            )
+
+            if not provider_config.saml_configuration:
+                self.stdout.write(f"[INFO] {provider_info} has no SAML configuration (may be intentional)")
+                null_config_count += 1
+                continue
 
-                    if current_config and current_config.id != provider_config.saml_configuration_id:
+            try:
+                current_config = SAMLConfiguration.current(
+                    provider_config.saml_configuration.site_id,
+                    provider_config.saml_configuration.slug
+                )
+
+                # Check for outdated configuration references
+                if current_config:
+                    if current_config.id != provider_config.saml_configuration_id:
                         self.stdout.write(
-                            f"Provider '{provider_config.slug}' (site {provider_config.site_id}) "
+                            f"[OUTDATED] {provider_info} "
                             f"has outdated config (ID: {provider_config.saml_configuration_id} -> {current_config.id})"
                         )
+                        outdated_count += 1
+                else:
+                    # No current config found - this might indicate the referenced config is no longer valid
+                    self.stdout.write(
+                        f"[WARNING] {provider_info} "
+                        f"references config (ID: {provider_config.saml_configuration_id}) but no current config "
+                        f"found for site {provider_config.saml_configuration.site_id}, "
+                        f"slug '{provider_config.saml_configuration.slug}'"
+                    )
+                    outdated_count += 1
+
+                if provider_config.saml_configuration.site_id != provider_config.site_id:
+                    config_site = provider_config.saml_configuration.site_id
+                    provider_site = provider_config.site_id
+                    self.stdout.write(
+                        f"[SITE_MISMATCH] {provider_info} "
+                        f"config site ({config_site}) != provider site ({provider_site})"
```

Comment:
1. The following is a proposed update.
2. If you want to keep all the previous indicators you had, maybe use `[WARNING] [SITE_MISMATCH] ...` or `[WARNING] Site mismatch: ...`.
3. The provider's site id will already be written, so it doesn't need to be repeated.
4. Would any additional info be needed to make it simpler to look into if you needed to manually fix? For example, `SAML config (id={config_id}, site_id={config_site})`? I don't know if this is needed. Think about how you would fix this warning.
```suggestion
                        f"[WARNING] {provider_info} "
                        f"SAML config site_id={config_site} does not match the provider's site_id."
```